pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_reconcile.go

```diff
@@ -112,7 +112,7 @@ func (c *controller) reconcile(ctx context.Context, apiBinding *apisv1alpha1.API
 		}
 
 		claimLogger.V(4).Info("listing resources")
-		objs, err := informer.Informer().GetIndexer().ByIndex(indexers.ByLogicalCluster, clusterName.String())
+		objs, err := informer.Lister().ByCluster(clusterName).List(labels.Everything())
```

Comment:
how can I check the new interface of a lister?
what is `ByCluster`? an index?
how does it differ from `configMapInformer.Lister().Cluster(cluster).ConfigMaps(namespace).Get(name)`?

Reply:
This was following the pattern of dynamic/generic listers doing `ByNamespace()` - but perhaps we should not do that and make it `.Cluster()`? You can see the implementation used in this PR already published to `kcp-dev/client-go` - https://github.com/kcp-dev/client-go/blob/272de79b2a7ab1d29e7d0db2d458258d035124d4/clients/metadata/metadatalister/shim.go#L54-L56

Reply:
yes, I'd prefer to have `Cluster()` method on the dynamic client